client/src/services/activity.ts

```diff
@@ -41,3 +42,55 @@ export const fetchPCStatus = async () => {
     return error;
   }
 };
+
+export const getRecentActivity = async () => {
+  const url = `${API_URL}/activity/all/recent`;
+  const settings = {
+    method: "GET",
+    headers: {
+      Accept: "application/json",
+      "Content-Type": "application/json",
+    },
+  };
+
+  try {
+    const response = await fetch(url, settings);
+    const data = await response.json();
+    return data;
+  } catch (error) {
+    return error;
+  }
+};
+
+export const useFetchActivities = () => {
+  const store = useBoundStore.getState();
+
+  useEffect(() => {
+    const fetchActivities = async () => {
+      try {
+        const activities = await getRecentActivity();
+
+        const activitiesWithProfiles = await Promise.all(
+          activities.map(async (activity) => {
+            const profile = await getGamerProfile(activity.student_number);
+
+            return {
+              first_name: profile.first_name,
+              last_name: profile.last_name,
+              ...activity,
+            };
```

Comment:
Now, that I think of it, idk if this was the best way to do this (sry for writing this in the issue requirements mb 😓!). 

I think it'll be easier if you edit the route [here](https://github.com/ubcesports/lounge-hub/blob/042bddb9859feb8012ca979bd09c4179267f46c9/app/routes/gamerActivity.js#L56) to be a SQL join on the gamer profiles to just get all the information in one API call. That way instead of calling the API 21 times, we just have to call it once.